### PR TITLE
[NFC][Support] Create helper function to parse bool.

### PR DESCRIPTION
**Pull request by jurahul**
- Create a helper template function to parse bool, to eliminate code duplication.

### CODE REVIEW

**Comment by jurahul** [comment]
Note, linux build failues look like infra issues: 

  | fatal: not a git repository (or any parent up to mount point /var/lib)

I don't have permission to rekick. Hoping it gets kicked when I update the branch again (and the infa issue resolve by that time)

File: llvm/lib/Support/CommandLine.cpp [hunk 1]
```diff
@@ -400,6 +400,22 @@ class CommandLineParser {
   SubCommand *LookupSubCommand(StringRef Name, std::string &NearestString);
 };
 
+template <typename T, T TrueVal, T FalseVal>
+bool parseBool(Option &O, StringRef ArgName, StringRef Arg, T &Value) {
```
Comment (MaskRay) [inline llvm/lib/Support/CommandLine.cpp:404]: move the function outside of the unnamed namespace and add `static` per https://llvm.org/docs/CodingStandards.html#anonymous-namespaces

The majority of commits omit the trailing full stop in the subject line.

**Review by MaskRay — APPROVED** [review]

File: llvm/lib/Support/CommandLine.cpp (hunk 1 above)
Comment (jurahul) [reply llvm/lib/Support/CommandLine.cpp]: Done.

**Comment by jurahul** [comment]
@MaskRay, can you please merge this PR?

File: llvm/lib/Support/CommandLine.cpp [hunk 2]
```diff
@@ -404,6 +404,22 @@ class CommandLineParser {
 
 static ManagedStatic<CommandLineParser> GlobalParser;
 
+template <typename T, T TrueVal, T FalseVal>
+static bool parseBool(Option &O, StringRef ArgName, StringRef Arg, T &Value) {
+  if (Arg == "" || Arg == "true" || Arg == "TRUE" || Arg == "True" ||
```
Comment (aganea) [inline llvm/lib/Support/CommandLine.cpp:409]: We could use `Arg.equals_insensitive("true") here. Same below for `false`

**Comment by jurahul** [comment]
But that would match more than what is in the code current. For example,
tRue or TrUE.

File: llvm/lib/Support/CommandLine.cpp (hunk 2 above)
Comment (jurahul) [reply llvm/lib/Support/CommandLine.cpp:409]: But that would match more than what the code current matches. For example, tRue or TrUE.

**Comment by jurahul** [comment]
@MaskRay or @joker-eph can one of you please merge the PR?

**Comment by jurahul** [comment]
Thanks!

**Pull request merged**